src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -21,6 +21,37 @@ public virtual DirectoryInfo Create(string path)
 
         #region UniversalTests
 
+        [Fact]
+        public void FileNameIsToString_NotFullPath()
+        {
+            // We're checking that we're maintaining the original path
+            RemoteInvoke(() =>
+            {
+                Environment.CurrentDirectory = TestDirectory;
+                string subdir = Path.GetRandomFileName();
+                DirectoryInfo info = Create(subdir);
+                Assert.Equal(subdir, info.ToString());
+            }).Dispose();
+        }
+
+        [Fact]
+        public void FileNameIsToString_FullPath()
+        {
+            string subdir = Path.GetRandomFileName();
+            string fullPath = Path.Join(TestDirectory, subdir);
+            DirectoryInfo info = Create(fullPath);
+            if (PlatformDetection.IsFullFramework)
+            {
+                // I think this was accidental. In Core we want to be consistent with constructing
+                // an Info manually then calling Create on it.
```

Comment:
I'm a little confused.  So currently the core behavior is the same as netfx, but you believe the netfx behavior to be a bug, and this PR is fixing that for core?  Is something breaking because of the current behavior?

Reply:
TLDR: I accidentally changed behavior from 2.0/NetFX and would rather _deliberately_ change the behavior to be consistent with other changes.

Currently the behavior is to put the normalized path into the original path, which is completely different than 2.0 and NetFX. NetFX and 2.0 will put the filename into the original path (and therefore the ToString()).

I could change the behavior to be the filename again, but didn't because:

1. When you create the DirectoryInfo first and then Create from there you get different behavior (better to be consistent)
2. This release was all about making consistency changes
3. It is slightly more risky to implement the file name behavior

Unfortunately we didn't have any tests around this behavior. I suspect that this variance was accidental. I found this when investigating Azure issues with this API.

Reply:
I don't think we should be deliberately changing behaviors at this point.  My preference would be to fix the regression for 2.1, and then decide if there's a meaningful change to make to behaviors for 2.2 and beyond.

Reply:
Note that the behavior of this change matches the documentation as well.  ToString() is supposed to be the original path.